File: giftastic-deployment-env/src/main/java/com/giftastic/giftastic/modules/cart/service/CartService.java
package com.giftastic.giftastic.modules.cart.service;

import java.util.List;
import java.util.UUID;

import com.giftastic.giftastic.modules.cart.dto.CartItemBulkRequest;
import com.giftastic.giftastic.modules.cart.dto.CartResponse;

public interface CartService {
    CartResponse getCart(UUID customerId);
    void addItem(UUID customerId, UUID productId, int quantity, String groupId, String metadata);
    void addItems(UUID customerId, List<CartItemBulkRequest> items);
    void updateQuantity(UUID customerId, UUID productId, String groupId, int quantity);
    void removeItem(UUID customerId, UUID productId, String groupId);
    void removeGroup(UUID customerId, String groupId);
    void clearCart(UUID customerId);
}

File: giftastic-frontend-deploy/frontend/src/services/cartService.js
import api from './api';

export const cartService = {
  async getCart(customerId) {
    const response = await api.get(`/cart/${customerId}`);
    return response.data;
  },

  async addItem(customerId, productId, quantity, groupId, metadata) {
    const params = { productId, quantity };
    if (groupId) {
      params.groupId = groupId;
    }
    if (metadata) {
      params.metadata = metadata;
    }

    const response = await api.post(`/cart/${customerId}/items`, null, {
      params
    });
    return response.data;
  },

  async addItems(customerId, items) {
    const response = await api.post(`/cart/${customerId}/items/bulk`, items);
    return response.data;
  },

  async updateItemQuantity(customerId, productId, quantity, groupId) {
    const params = { quantity };
    if (groupId) {
      params.groupId = groupId;
    }

    const response = await api.patch(`/cart/${customerId}/items/${productId}`, null, {
      params
    });
    return response.data;
  },

  async removeItem(customerId, productId, groupId) {
    const response = await api.delete(`/cart/${customerId}/items/${productId}`, {
      params: groupId ? { groupId } : undefined
    });
    return response.data;
  },

  async removeGroup(customerId, groupId) {
    const response = await api.delete(`/cart/${customerId}/groups/${groupId}`);
    return response.data;
  },

  async clearCart(customerId) {
    const response = await api.delete(`/cart/${customerId}/clear`);
    return response.data;
  }
};
